source/server/server.cc

```diff
@@ -292,6 +293,46 @@ void loadBootstrap(absl::optional<uint32_t> bootstrap_version,
     throw EnvoyException(fmt::format("Unknown bootstrap version {}.", *bootstrap_version));
   }
 }
+
+bool canBeRegisteredAsInlineHeader(const Http::LowerCaseString& header_name) {
+  // 'set-cookie' cannot currently be registered as an inline header.
+  if (header_name == Http::Headers::get().SetCookie) {
+    return false;
+  }
+  return true;
+}
+
+void registerCustomInlineHeadersFromBootstrap(
+    const envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+  for (const auto& inline_header : bootstrap.inline_headers()) {
+    const Http::LowerCaseString lower_case_name(inline_header.inline_header_name());
+    if (!canBeRegisteredAsInlineHeader(lower_case_name)) {
+      throw EnvoyException(fmt::format("Header {} cannot be registered as an inline header.",
+                                       inline_header.inline_header_name()));
+    }
+    switch (inline_header.inline_header_type()) {
+    case envoy::config::bootstrap::v3::CustomInlineHeader::REQUEST_HEADER:
```

Comment:
Can one add as a custom inline header a header which is already an inline header?  I think it's worth a unit test for this.

Reply:
> Can one add as a custom inline header a header which is already an inline header?
Yes.

> I think it's worth a unit test for this.

I think this scenario has been verified in the test of `CustomInlineHeaderRegistry::registerInlineHeader` itself. We can register the header with the same name as the inline header repeatedly without any negative effects. 

https://github.com/envoyproxy/envoy/blob/d483098022e72de39a23dc5f9ed65a044ae86b3a/test/common/http/header_map_impl_test.cc#L365-L368

https://github.com/envoyproxy/envoy/blob/d483098022e72de39a23dc5f9ed65a044ae86b3a/test/common/http/header_map_impl_test.cc#L389-l399